treewide: strip trailing whitespace

namiltd opened the pull request:
Strip trailing whitespace in all code:
```
find . -type f | grep "\.c$" | xargs sed -i 's/[ \t]\+$//'
find . -type f | grep "\.h$" | xargs sed -i 's/[ \t]\+$//'
find . -type f | grep "\.dts$" | xargs sed -i 's/[ \t]\+$//'
find . -type f | grep "\.dtsi$" | xargs sed -i 's/[ \t]\+$//'
```

namiltd commented:
Code cleaning from unnecessary characters for 203 files done.

namiltd commented:
Ping @robimarko

KanjiMonster commented on target/linux/bcm27xx/patches-6.6/950-0547-ASOC-dwc-Improve-DMA-shutdown.patch:
```diff
@@ -13,15 +13,6 @@ Signed-off-by: Phil Elwell <phil@raspberrypi.com>
 
 --- a/sound/soc/dwc/dwc-i2s.c
```
Please squash this into the first commit, don't break stuff just to fix it later.

namiltd replied:
The first commit is just a result of sed runs.
Thanks to the separation, you can see what exactly has changed.
The second commit removes what is no longer needed or requires updating due to changes.
Squashing will make it so no one will be able to verify it.

KanjiMonster replied:
Each individual commit needs to be correct and not break compilation. If the patches need rebuilding after running the whitespace changes, then these changes need to be in the same commit as the whitespace changes.

namiltd replied:
Patches squashed. I left the commit split in bd796b56b1a8fac9120917530c81070cd0b52f2d and 48236d0b91eb949ed1f787c4019307d8c81eee60 for change analysis.

KanjiMonster commented:
While cleaning up white space in non-patch files is fine and good, I'm a bit unsure about patch files, especially if those patches weren't from us but imported from somewhere else.

E.g. the bcm27xx patches were taken from https://github.com/raspberrypi/linux/, so the next time these patches get updated, all trailing space will just get re-added. And if someone wants to just add a commit from there it may now not apply anymore due to the white space changes.

namiltd commented:
Patches in bcm26xx are added with the following command:
e.g: `git format-patch -N v6.6.85..HEAD`
So only new patches are added.

namiltd commented:
If this PR were to be approved, now would be the best time (before kernel 6.12 targets are approved).

namiltd commented:
I see that the chances of approval this are slim.

namiltd commented:
Ok, I will limit the patch.

namiltd committed:
treewide: strip trailing whitespace

Strip trailing whitespace in all code:
find . -type f | grep "\.c$" | xargs sed -i 's/[ \t]\+$//'
find . -type f | grep "\.h$" | xargs sed -i 's/[ \t]\+$//'
find . -type f | grep "\.dts$" | xargs sed -i 's/[ \t]\+$//'
find . -type f | grep "\.dtsi$" | xargs sed -i 's/[ \t]\+$//'

Signed-off-by: Mieczyslaw Nalewaj <namiltd@yahoo.com>
Link: https://github.com/openwrt/openwrt/pull/18626
Signed-off-by: Robert Marko <robimarko@gmail.com>

namiltd commented:
PR is ready, all CI tests are OK.
